### ISSUE TITLE
Missing documentation for simply using tensors and optimizers together

### ISSUE THREAD

**Issue by ifsheldon** [LaurentMazare/tch-rs#965]
Hi I'm new to tch-rs. I want to start simple like doing gradient descent on tensors, not network weights. However, I cannot find a way to do it in the documentation.

I figured this out, but it took me a while. Here is my example:
```rust
use tch::{Device, Kind, Tensor};
use tch::nn::{Adam, OptimizerConfig, VarStore};

fn main() {
    let vs = VarStore::new(Device::Cpu);
    let root = vs.root();
    let mut adam = Adam::default().build(&vs, 1e-1).unwrap();
    let v = root.randn("v", &[2, 2, 2, 2], 0.0, 1.0);
    let v_copy = v.copy();
    v_copy.print();
    let loss = v.pow_tensor_scalar(2).sum(Kind::Float);
    loss.backward();
    adam.step();
    adam.zero_grad();
    v.print();
    assert!(!v.allclose(&v_copy, 1e-5, 1e-5, false));
}
```

I can make a PR if you like.
